addons/account/models/partner.py

```diff
@@ -370,7 +370,7 @@ def _compute_has_unreconciled_entries(self):
 
     def mark_as_reconciled(self):
         self.env['account.partial.reconcile'].check_access_rights('write')
-        return self.sudo().with_context(company_id=self.env.company.id).write({'last_time_entries_checked': time.strftime(DEFAULT_SERVER_DATETIME_FORMAT)})
+        return self.sudo().write({'last_time_entries_checked': time.strftime(DEFAULT_SERVER_DATETIME_FORMAT)})
```

Comment:
Should be safe to remove as the root cause was the old behavior of sudo(). Cfr OPW-1770018 and #19274 and related commits.